src/org/parosproxy/paros/extension/option/OptionsParamCertificate.java

```diff
@@ -61,14 +72,75 @@ public OptionsParamCertificate() {
     @Override
     protected void parse() {
 
-        // always turn off client cert
-        setUseClientCert(false);
-        setClientCertLocation("");
+        clientCertCheck();
+        saveClientCertSettings();
         
         allowUnsafeSslRenegotiation = getBoolean(ALLOW_UNSAFE_SSL_RENEGOTIATION, false);
         setAllowUnsafeSslRenegotiationSystemProperty(allowUnsafeSslRenegotiation);
     }
 
+    /**
+     *  Saves the client cert settings if the flag is set explicitly.
+     *  Only works for the CLI currently.
+     */
+    private void saveClientCertSettings(){
+
+        if (getBoolean(PERSIST_CLIENT_CERT, false)){
+            logger.warn("Saving Client Certificate settings: password will be found in config");
+            setUseClientCert(getBoolean(USE_CLIENT_CERT, false));
+            setClientCertLocation(getString(CLIENT_CERT_LOCATION, ""));
+            setClientCertPassword(getString(CLIENT_CERT_PASSWORD, ""));
+            setClientCertIndex(getInt(CLIENT_CERT_INDEX, 0));
+
+        } else {
+            // Default to clear settings
+            setUseClientCert(false);
+            setClientCertLocation("");
+            setClientCertPassword("");
+            setClientCertIndex(0);
+        }
+    }
+
+    /**
+     * Enables ClientCertificate from -config CLI parameters
+     * Requires location, password and a flag to use client certificate.
+     */
+    private void clientCertCheck(){
+
+        boolean enableClientCert = getBoolean(USE_CLIENT_CERT, false);
+        String certPath = getString(CLIENT_CERT_LOCATION, "");
+        String certPass = getString(CLIENT_CERT_PASSWORD, "");
+        int certIndex = getInt(CLIENT_CERT_INDEX, 0);
+
+        if(enableClientCert && !certPath.isEmpty() && !certPass.isEmpty()) {
+            try {
+
+                SSLContextManager contextManager = getSSLContextManager();
+                int ksIndex = contextManager.loadPKCS12Certificate(certPath, certPass);
+                contextManager.unlockKey(ksIndex, certIndex, certPass);
+                contextManager.setDefaultKey(ksIndex, certIndex);
+
+                setActiveCertificate();
+                setEnableCertificate(true);
+
+                logger.info("Client Certificate enabled from CLI");
+                logger.info("Use -config certificate.persist=1 to save settings");

```

Comment:
This needs update, `true` instead of `1`. We could also omit the log if already persisting?